### PR TITLE
More explanatory about box.

### PR DESCRIPTION
**Pull request by baconpaul**
This initial commit rewires the about box so the text painted on screen
comes from code, and also includes a build date and time. See issue #157 for
a discussion of the final state; this is an interim step to at least get correct information
and a build date into our releases.

### CODE REVIEW

File: src/common/gui/CAboutBox.cpp [hunk 1]
```diff
@@ -29,6 +35,26 @@ void CAboutBox::draw(CDrawContext* pContext)
    if (value)
    {
       _aboutBitmap->draw(pContext, getViewSize(), CPoint(0, 0), 0xff);
+
+      int strHeight = infoFont->getSize(); // There should really be a better API for this in VSTGUI
+      std::vector< std::string > msgs = { {
+              std::string( "Version 1.6.0 beta (build: " ) + __DATE__ + " " + __TIME__ + ")",
+              "Released under the GNU General Public License, v3",
```
Comment (jsakkine) [inline src/common/gui/CAboutBox.cpp]: Should be a file lets say `VERSION` in root directory where the version would be grabbed?

File: src/common/gui/CAboutBox.cpp [hunk 2]
```diff
@@ -29,6 +35,26 @@ void CAboutBox::draw(CDrawContext* pContext)
...
+              "VST Plug-in technology by Steinberg, AU Plugin Technology by Apple Computer"
+          } };
+
+      int yMargin = 5;
+      int yPos = toDisplay.getHeight() - msgs.size() * (strHeight + yMargin); // one for the last; one for the margin
+      int xPos = strHeight;
+      pContext->setFontColor( kWhiteCColor );
+      pContext->setFont( infoFont );
+      for( auto s : msgs )
+      {
```
Comment (jsakkine) [inline src/common/gui/CAboutBox.cpp]: Cosmetic: hinking about the white-spacing here and also in function calls. I use a different syntax. Which one Surge originally uses?

File: src/common/gui/CAboutBox.cpp [hunk 3]
```diff
@@ -92,4 +118,4 @@ void CAboutBox::unSplash()
          getFrame()->setModalView(NULL);
       }
    }
-}
\ No newline at end of file
+}
```
Comment (jsakkine) [inline src/common/gui/CAboutBox.cpp]: Yep looking at `setModalView()` call, it uses a different as your commit.

File: src/common/gui/CAboutBox.cpp (hunk 1 above)
Comment (baconpaul) [reply src/common/gui/CAboutBox.cpp]: Well we should have a built artifact with all our information! That's why this is an incomplete diff. For now I just hardcoded 1.6 so I didn't have to deal with cross platform resource loads but to properly finish this we need to create the XML file we were discussing which should include the version and so on and then not have this here.

My thinking was "hardcoded as string is better than as png" and "getting the build date and correct license" was valuable which is why I did this slightly but still slightly less tacky interim step.

File: src/common/gui/CAboutBox.cpp (hunk 2 above)
Comment (baconpaul) [reply src/common/gui/CAboutBox.cpp]: You are 100% correct. That convention will never leave my fingers. I will go and fix it and squash in.

Comment (baconpaul) [reply src/common/gui/CAboutBox.cpp]: OK I pushed this change. Thanks for the careful look!

**Pull request merged**